src/main/java/com/mraof/minestuck/data/dialogue/CarapacianSoldierDialogue.java

```diff
@@ -0,0 +1,108 @@
+					.node(new NodeBuilder(l.defaultKeyMsg("They explain that Dreamers are heroes who reside in towers on the moons of Prospit and Derse.")))
+					.node(new NodeBuilder(l.defaultKeyMsg("There is one for each Land that exists, although they don't know what the connection is there.")))
+			);
+			
+			builder.addStart(new NodeSelectorBuilder()
+					.node(hasEntered(), new NodeBuilder(dots).addDescription(l.subMsg("familiar", "They look at you with surprise, saying you remind them of the Dreamers."))
+							.addResponse(new ResponseBuilder(l.subMsg("who", "Who are the Dreamers?"))
+									.nextDialogue(explainDreamers)))
+					.defaultNode(new NodeBuilder(dots).addDescription(l.subMsg("unfamiliar", "They are curious as to who you are and what you are doing here."))));
+		}));
+		
+		provider.addRandomlySelectable("not_so_bad", defaultWeight(isInSkaia()), new ChainBuilder()
+				.node(new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They recount one time where they went out to an isolated part of the Battlefield. Unexpectedly, a member of enemy scouting party came across them while sat on a hill side.")))
+				.node(new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They went to draw their weapon only to realize they forgot it back at camp. The two soldiers stared at each other for what felt like ages.")))
+				.node(new NodeBuilder(dots).addDescription(l.defaultKeyMsg("To make things worse, the rest of the scouting party called out to see what was going on. But just when they thought their life was coming to an end, the other soldier pretended like they weren't there and announced that the area was clear.")))
+				.node(new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They never ran so fast, trying to get back somewhere safe. And they certainly never leave anywhere without a weapon now.")))
+		);
+		
+		provider.addRandomlySelectable("enemy", defaultWeight(all(isInSkaia(), isProspitian())),
+				new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They are panicking about whether any Dersites are nearby.")));
+		
+		provider.addRandomlySelectable("sword_barter", weighted(40, isHolding(MSItems.REGISWORD.get())), new FolderedDialogue(builder ->
+		{
+			var goodbye = l.subMsg("goodbye", "That's really strange. Good luck with that!");
+			
+			var thanks = builder.add("thanks", new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They thank you for helping them.")));
+			
+			var barter = builder.add("barter", new NodeBuilder(dots).addDescription(l.defaultKeyMsg("They ask if you have a new sword for them."))
+					.addResponse(new ResponseBuilder(l.subMsg("give_item", "[Give them the %s]", DialogueMessage.Argument.MATCHED_ITEM))
+							.addPlayerMessage(l.subMsg("give_item.reply", "Here you go!"))
+							.visibleCondition(resource -> "Must have a sword and must not be carrying a regisword", all(new Condition.ItemTagMatch(ItemTags.SWORDS), none(new Condition.PlayerHasItem(MSItems.REGISWORD.get(), 1))))
```

Comment:
I made it so that they cant be carrying a regisword because I wanted to prevent players from giving another one back to them and didnt want to create an item tag just for this dialogue. Im open to suggestions as it is slightly awkward not to allow a regisword to be in the players inventory at all

Reply:
The simplest way of solving this is to make a special case of `Condition.ItemTagMatch` that specifically matches against non-regisword swords. It doesn't have to immediately be reusable for other dialogue, and we can come up with a suitable generalization when we want something like that.

Reply:
Im not sure I understand, specifically because I am already using an ItemTagMatch Condition to check for swords

Reply:
What you're using checks for swords. What you want to is checking for non-regisword swords, right? If it's difficult to create an item tag for that, you could create a specialized condition instead.

Reply:
Ah didnt realize you were suggesting the creation of a new Condition. How about the name of ItemTagMatchExclude?

Reply:
Sure